syntax = "proto3";
package Service.Proto;

message Home
{
	string      GamePath    = 1;
    LaunchGame  LaunchGame  = 2;
    Island      Island      = 3;
    int32       SelectedIndex = 4;
}

message LaunchGame
{
    bool            UsingHoyolabAccount               =  1;
    bool            AreCommandLineArgumentsEnabled    =  2;
    bool            IsWindowsHDREnabled               =  3;
    bool            IsFullScreen                      =  4;
    bool            IsExclusive                       =  5;
    bool            IsBorderless                      =  6;
    bool            IsScreenWidthEnabled              =  7;
    int32           ScreenWidth                       =  8;
    bool            IsScreenHeightEnabled             =  9;
    int32           ScreenHeight                      = 10;
}

message Island
{

	bool            EnableSetFieldOfView              =  1;
	float           FieldOfView                       =  2;
    bool            FixLowFovScene                    =  3;
    bool            DisableFog                        =  4;
    bool            EnableSetTargetFrameRate          =  5;
    uint32          TargetFrameRate                   =  6;
    bool            RemoveOpenTeamProgress            =  7;
    bool            HideQuestBanner                   =  8;
    bool            DisableEventCameraMove            =  9;
    bool            DisableShowDamageText             = 10;
    bool            UsingTouchScreen                  = 11;
    bool            RedirectCombineEntry              = 12;
    map<int32,bool> items                             = 13;

}

message AppSettings
{
    bool            StealthMode                       =  1;
    bool            CloseAsQuit                       =  2;
    bool            RestrictedTokens                  =  3;
    int32           Lang                              =  4;
    bool            LangOverride                      =  5;
}

message PrivateVars
{
    int64           WindowRect                        =  1;
    int32		    StartCount                        =  2;
    bool            CollapseNavView                   =  3;
    int32		    Version                           =  4;   
    AppEdition      AppEdition                        =  5;
}

enum AppEdition 
{
    APP_EDITION_UNSPECIFIED = 0;
    APP_EDITION_LITE        = 1;
    APP_EDITION_STANDARD    = 2;
    APP_EDITION_Dev         = 3;
}

message Settings
{
    Home                          Home        = 1;
    AppSettings                   AppSettings = 2;
    PrivateVars                   PrivateVars = 3;
}